lib/spack/spack/reporters/cdash.py

```diff
@@ -51,39 +56,57 @@ class CDash(Reporter):
     CDash instance hosted at https://mydomain.com/cdash.
     """
 
-    def __init__(self, install_command, cdash_upload_url):
-        Reporter.__init__(self, install_command, cdash_upload_url)
+    def __init__(self, args):
+        Reporter.__init__(self, args)
         self.template_dir = os.path.join('reports', 'cdash')
-        self.hostname = socket.gethostname()
+        self.cdash_upload_url = args.cdash_upload_url
+        self.install_command = ' '.join(args.package)
+        if args.cdash_build:
+            self.buildname = args.cdash_build
+        else:
+            self.buildname = self.install_command
+        if args.cdash_site:
+            self.site = args.cdash_site
+        else:
+            self.site = socket.gethostname()
         self.osname = platform.system()
-        self.starttime = int(time.time())
-        # TODO: remove hardcoded use of Experimental here.
-        #       Make the submission model configurable.
-        self.buildstamp = time.strftime("%Y%m%d-%H%M-Experimental",
-                                        time.localtime(self.starttime))
+        self.endtime = int(time.time())
+        buildstamp_format = "%Y%m%d-%H%M-{0}".format(args.cdash_track)
+        self.buildstamp = time.strftime(buildstamp_format,
+                                        time.localtime(self.endtime))
+        self.buildId = None
+        self.revision = ''
+        git = which('git')
+        with working_dir(spack.paths.spack_root):
+            self.revision = git('rev-parse', 'HEAD', output=str).strip()
 
     def build_report(self, filename, report_data):
         self.initialize_report(filename, report_data)
 
         for phase in cdash_phases:
             report_data[phase] = {}
-            report_data[phase]['log'] = ""
+            report_data[phase]['log'] = cStringIO()
```

Comment:
`cStringIO` doesn't exist in Python 3 -- this just gets you an `io.StringIO` since you imported it from `six.moves`.  To me more forward compatible, can you use `from six import StringIO` instead?  Or did you see serious improvements with `cStringIO`?  Both are just `io.StringIO` in python 3.